src/Microsoft.DotNet.Cli.Utils/BuiltInCommand.cs

```diff
@@ -100,6 +108,29 @@ public ICommand OnErrorLine(Action<string> handler)
             return this;
         }
 
+        private class BuiltInCommandEnvironment : IBuiltInCommandEnvironment
```

Comment:
Would it make sense to move this over to our InternalAbstractions?

I like the console abstraction, and think it could be useful in other places.